src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -189,19 +200,21 @@ public PolicyEntity call() throws Exception {
 
             @Override
             public List<String> call() throws Exception {
-                CriteriaBuilder cb = em.getCriteriaBuilder();
-                CriteriaQuery<PolicyEntity> q = cb.createQuery(PolicyEntity.class);
-                Root<PolicyEntity> r = q.from(PolicyEntity.class);
-                q.select(r).where(cb.and(cb.equal(r.get("domain").get("Id"), domainid)));
-                List<String> policy = new ArrayList<String>();
-                List<PolicyEntity> PolicyList = em.createQuery(q).getResultList();
-                if (PolicyList.isEmpty() == false) {
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+                CriteriaQuery<PolicyEntity> query = criteriaBuilder.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = query.from(PolicyEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("domain").get("Id"), domainid)));
+                List<String> policy = null;
+                List<PolicyEntity> PolicyList = em.createQuery(query).getResultList();
+                if (!PolicyList.isEmpty()) {
+                    policy = new ArrayList<String>();
                     for (PolicyEntity mgrPolicy : PolicyList) {
                         policy.add(new String(mgrPolicy.getId()));
                     }
                     return policy;
                 } else {
-                    return null;
+                    throw new Exception("Domain Entity does not exists...");
```

Comment:
this is not true, this can also be empty if the domain does not have any policies under it right? I think in this case we should just return an empty list.

Reply:
ok